tests/tests_geomstats/test_hypersphere.py

```diff
@@ -1,911 +1,701 @@
-        regularized_norm_expected = gs.mod(norm_expected, 2 * gs.pi)
-        expected = expected / norm_expected * regularized_norm_expected
-
-        # The Log can be the opposite vector on the tangent space,
-        # whose Exp gives the base_point
-        are_close = gs.allclose(result, expected)
-        norm_2pi = gs.isclose(gs.linalg.norm(result - expected), 2 * gs.pi)
-        self.assertTrue(are_close or norm_2pi)
-
-    def test_exp_and_log_and_projection_to_tangent_space_edge_case(self):
-        """Test Log and Exp.
-
-        Test that the Riemannian exponential
-        and the Riemannian logarithm are inverse.
-
-        Expect their composition to give the identity function.
-
-        NB: points on the n-dimensional sphere are
-        (n+1)-D vectors of norm 1.
-        """
-        # Riemannian Exp then Riemannian Log
-        # Edge case: tangent vector has norm < epsilon
-        base_point = gs.array([10.0, -2.0, -0.5, 34.0, 3.0])
-        base_point = base_point / gs.linalg.norm(base_point)
-        vector = 1e-4 * gs.array([0.06, -51.0, 6.0, 5.0, 3.0])
-        vector = self.space.to_tangent(vector=vector, base_point=base_point)
-
-        exp = self.metric.exp(tangent_vec=vector, base_point=base_point)
-        result = self.metric.log(point=exp, base_point=base_point)
-        self.assertAllClose(result, vector)
-
-    def test_squared_norm_and_squared_dist(self):
-        """
-        Test that the squared distance between two points is
-        the squared norm of their logarithm.
-        """
-        point_a = 1.0 / gs.sqrt(129.0) * gs.array([10.0, -2.0, -5.0, 0.0, 0.0])
-        point_b = 1.0 / gs.sqrt(435.0) * gs.array([1.0, -20.0, -5.0, 0.0, 3.0])
-        log = self.metric.log(point=point_a, base_point=point_b)
-        result = self.metric.squared_norm(vector=log)
-        expected = self.metric.squared_dist(point_a, point_b)
-
+    def test_extrinsic_to_angle(self, dim, point, expected):
+        space = self.space(dim)
+        result = space.extrinsic_to_angle(point)
         self.assertAllClose(result, expected)
 
-    def test_squared_dist_vectorization_single_sample(self):
-        one_point_a = self.space.random_uniform()
-        one_point_b = self.space.random_uniform()
-
-        result = self.metric.squared_dist(one_point_a, one_point_b)
-        self.assertAllClose(gs.shape(result), ())
-
-        one_point_a = gs.to_ndarray(one_point_a, to_ndim=2)
-        result = self.metric.squared_dist(one_point_a, one_point_b)
-        self.assertAllClose(gs.shape(result), (1,))
-
-        one_point_b = gs.to_ndarray(one_point_b, to_ndim=2)
-        result = self.metric.squared_dist(one_point_a, one_point_b)
-        self.assertAllClose(gs.shape(result), (1,))
-
-        one_point_a = self.space.random_uniform()
-        result = self.metric.squared_dist(one_point_a, one_point_b)
-        self.assertAllClose(gs.shape(result), (1,))
-
-    def test_squared_dist_vectorization_n_samples(self):
```

Comment:
missing? this could become a test in the RiemannianMetricParametrizer (squared_dist_shape)

Reply:
already part of improvs